rasa/nlu/registry.py

```diff
@@ -139,15 +139,29 @@ def get_component_class(component_name: Text) -> Type["Component"]:
         if component_name not in old_style_names:
             try:
                 return class_from_module_path(component_name)
-            except Exception:
+            except ModuleNotFoundError as e:
+                # when component_name is a path to a class but that path is invalid
                 raise Exception(
-                    "Failed to find component class for '{}'. Unknown "
-                    "component name. Check your configured pipeline and make "
-                    "sure the mentioned component is not misspelled. If you "
-                    "are creating your own component, make sure it is either "
-                    "listed as part of the `component_classes` in "
-                    "`rasa.nlu.registry.py` or is a proper name of a class "
-                    "in a module.".format(component_name)
+                    "Failed to find component class for '{}'.Unknown component name.\n{}".format(
+                        component_name, e.msg
+                    )
+                )
+            except AttributeError:
+                # when component_name is a path to a class but the path does not contain that class
+                module_name, _, class_name = component_name.rpartition(".")
+                raise Exception(
+                    "Failed to find component class for '{}'.Unknown component name.\n"
+                    "Cannot find class '{}' in module {}.".format(
+                        component_name, class_name, module_name
+                    )
+                )
+            except ImportError:
+                # when component_name is a class name and not part of old_style_names
+                raise Exception(
+                    "Failed to find component class for '{0}'.Unknown component name.\n"
```

Comment:
I think it's okay if we just keep the second part. Can we provide any solution in the exception message or a link to some explanations?
```suggestion
                    "Cannot import class '{0}' from global namespace.\n"
```

Reply:
What about something like?

`"Cannot import class '{0}' from global namespace.\nCheck that you have not made a typo in the class name and that you have installed the correct package."
`

Reply:
Sounds good!

Reply:
@wochinge I don't think the explanation is correct. 

That exception is triggered when the class cannot be retrieved from globals i.e.

`module = globals().get(module_path, locals().get(module_path))`

fails and where `module_path` is the class name.


So I think the message should be:

`Cannot find class '{0}' from global namespace. Check that you have not made a typo in the class name and that you have imported the class into the global namespace.`

But I am not sure at what point the user will have a chance to manually import the class into the global namespace.

Thoughts?

Also, the unit tests I've seen so far has `lookup_path` set to `None`.  Do you know what are the circumstances when `lookup_path` is not `None`?

Reply:
> Also, the unit tests I've seen so far has lookup_path set to None. Do you know what are the circumstances when lookup_path is not None?

which `lookup_path`? 

> Cannot find class '{0}' from global namespace. Check that you have not made a typo in the class name and that you have imported the class into the global namespace.

Ok 👍 
Let's do `Cannot find class '{0}' from global namespace. Please check that there is no typo in the class name and that you have imported the class into the global namespace.

Reply:
> which lookup_path?

I'm referring to the `lookup_path` in [this ](https://github.com/RasaHQ/rasa/blob/e5f2375d3916fdc4ab568ac45e03f9c67bd2f137/rasa/utils/common.py#L163-L187)function's argument.

> Ok 👍
> Let's do `Cannot find class '{0}' from global namespace. Please check that there is no typo in the class name and that you have imported the class into the global namespace.

Done!